# Task
Load the Iris dataset and generate its numerical, categorical, and complete descriptive statistics. Summarize key observations from each statistics table.

## Cargar Dataset de Iris

### Subtask:
Cargar el conjunto de datos de Iris desde una biblioteca de Python (por ejemplo, scikit-learn o seaborn) en un DataFrame de pandas.


**Reasoning**:
The subtask requires loading the Iris dataset from `sklearn.datasets`, converting it into a pandas DataFrame, adding the species column using the target names, and then displaying the first few rows to verify the successful loading. All these steps can be performed in a single code block.



In [1]:
import pandas as pd
from sklearn.datasets import load_iris

# 1. Cargar el dataset de Iris
iris = load_iris()

# 2. Crear un DataFrame de pandas
df_iris = pd.DataFrame(data=iris.data, columns=iris.feature_names)

# 3. Añadir la columna de especies
df_iris['species'] = iris.target_names[iris.target]

# 4. Mostrar las primeras filas del DataFrame
print("Dataset de Iris cargado exitosamente:")
df_iris.head()

Dataset de Iris cargado exitosamente:


,sepal length (cm),sepal width (cm),petal length (cm),petal width (cm),species
0,5.1,3.5,1.4,0.2,setosa
1,4.9,3.0,1.4,0.2,setosa
2,4.7,3.2,1.3,0.2,setosa
3,4.6,3.1,1.5,0.2,setosa
4,5.0,3.6,1.4,0.2,setosa


## Estadísticas Numéricas Descriptivas

### Subtask:
Calcular y mostrar las estadísticas descriptivas (conteo, media, desviación estándar, mínimo, cuartiles, máximo) para todas las columnas numéricas del dataset de Iris. Presentar los resultados en una tabla separada.


**Reasoning**:
The subtask requires calculating and displaying descriptive statistics for numerical columns. I will select the numerical columns from `df_iris` using `select_dtypes(include='number')`, then apply the `.describe()` method, and finally print the result.



In [2]:
print("Estadísticas Descriptivas Numéricas del Dataset de Iris:")
df_iris.select_dtypes(include='number').describe()

Estadísticas Descriptivas Numéricas del Dataset de Iris:


,sepal length (cm),sepal width (cm),petal length (cm),petal width (cm)
count,150.000000,150.000000,150.000000,150.000000
mean,5.843333,3.057333,3.758000,1.199333
std,0.828066,0.435866,1.765298,0.762238
min,4.300000,2.000000,1.000000,0.100000
25%,5.100000,2.800000,1.600000,0.300000
50%,5.800000,3.000000,4.350000,1.300000
75%,6.400000,3.300000,5.100000,1.800000
max,7.900000,4.400000,6.900000,2.500000


### Observaciones Clave de las Estadísticas Numéricas:

*   **Longitud del sépalo (sepal length):**
    *   El promedio es de 5.84 cm, con una desviación estándar de 0.83 cm, lo que indica una variabilidad moderada.
    *   Los valores van desde 4.3 cm hasta 7.9 cm.
*   **Ancho del sépalo (sepal width):**
    *   El promedio es de 3.06 cm, con una desviación estándar menor (0.44 cm), sugiriendo menos dispersión en este atributo.
    *   Los valores oscilan entre 2.0 cm y 4.4 cm.
*   **Longitud del pétalo (petal length):**
    *   Tiene el rango más amplio de valores (1.0 cm a 6.9 cm) y la mayor desviación estándar (1.77 cm), lo que implica una gran variabilidad, probablemente debido a las diferencias entre especies.
    *   La media es de 3.76 cm.
*   **Ancho del pétalo (petal width):**
    *   Similar a la longitud del pétalo, presenta una alta variabilidad con una desviación estándar de 0.76 cm y un rango de 0.1 cm a 2.5 cm.
    *   La media es de 1.20 cm.

En general, las longitudes y anchos de los pétalos muestran una mayor variabilidad en comparación con los sépalos, lo que podría ser un indicador importante para distinguir entre las diferentes especies de Iris.

## Estadísticas Categóricas Descriptivas

### Subtask:
Calcular y mostrar las estadísticas descriptivas (conteo, valor único, moda, frecuencia) para todas las columnas categóricas del dataset de Iris. Presentar los resultados en una tabla separada.


**Reasoning**:
The subtask requires identifying categorical columns, calculating their descriptive statistics, and displaying them. The `describe(include='object')` method is suitable for this purpose, as 'species' is the only categorical column.



In [3]:
print("Estadísticas Descriptivas Categóricas del Dataset de Iris:")
df_iris.select_dtypes(include='object').describe()

Estadísticas Descriptivas Categóricas del Dataset de Iris:


,species
count,150
unique,3
top,setosa
freq,50


### Observaciones Clave de las Estadísticas Categóricas:

*   **Columna 'species':**
    *   El dataset contiene 150 registros en total.
    *   Existen 3 categorías únicas de especies de Iris: 'setosa', 'versicolor' y 'virginica'.
    *   La especie más frecuente ('top') es 'setosa', con una frecuencia ('freq') de 50. Esto implica que cada una de las tres especies ('setosa', 'versicolor', 'virginica') tiene exactamente 50 entradas en el dataset, indicando un balance perfecto entre las clases.

## Estadísticas Completas Descriptivas

### Subtask:
Calcular y mostrar las estadísticas descriptivas completas (incluyendo numéricas y categóricas) para todas las columnas del dataset de Iris. Presentar los resultados en una tabla separada.


**Reasoning**:
The subtask requires calculating and displaying complete descriptive statistics for all columns, including both numerical and categorical. The `describe(include='all')` method is appropriate for this purpose.



In [5]:
print("Estadísticas Descriptivas Completas del Dataset de Iris:")
df_iris.describe(include='all')

Estadísticas Descriptivas Completas del Dataset de Iris:


,sepal length (cm),sepal width (cm),petal length (cm),petal width (cm),species
count,150.000000,150.000000,150.000000,150.000000,150
unique,NaN,NaN,NaN,NaN,3
top,NaN,NaN,NaN,NaN,setosa
freq,NaN,NaN,NaN,NaN,50
mean,5.843333,3.057333,3.758000,1.199333,NaN
std,0.828066,0.435866,1.765298,0.762238,NaN
min,4.300000,2.000000,1.000000,0.100000,NaN
25%,5.100000,2.800000,1.600000,0.300000,NaN
50%,5.800000,3.000000,4.350000,1.300000,NaN
75%,6.400000,3.300000,5.100000,1.800000,NaN


### Observaciones Clave de las Estadísticas Completas:

La tabla de estadísticas descriptivas completas `df_iris.describe(include='all')` proporciona un resumen exhaustivo de todas las columnas, combinando las observaciones numéricas y categóricas previamente notadas:

*   **Columnas Numéricas (sepal length, sepal width, petal length, petal width):**
    *   Las filas `mean`, `std`, `min`, `25%`, `50%`, `75%`, `max` proporcionan un resumen de la tendencia central, dispersión y distribución de cada característica numérica.
    *   `count` es 150 para todas las columnas, confirmando la ausencia de valores nulos.
    *   `petal length` y `petal width` muestran una mayor variabilidad (desviaciones estándar más altas y rangos más amplios) en comparación con `sepal length` y `sepal width`. Esto sugiere que las dimensiones de los pétalos son características más distintivas entre las especies.

*   **Columna Categórica (species):**
    *   `count` es 150, como se esperaba.
    *   `unique` es 3, indicando que hay tres especies distintas en el dataset.
    *   `top` es 'setosa', lo que significa que 'setosa' es el valor más frecuente.
    *   `freq` es 50, confirmando que cada una de las tres especies ('setosa', 'versicolor', 'virginica') está igualmente representada con 50 instancias, lo que indica un dataset balanceado.

## Final Task

### Subtask:
Proporcionar un resumen de las estadísticas descriptivas de las columnas numéricas, categóricas y completas del dataset de Iris, destacando cualquier observación clave de cada tabla.


## Summary:

### Data Analysis Key Findings

*   The Iris dataset was successfully loaded, containing 150 records and no missing values across all features.
*   **Numerical Features:**
    *   **Sepal Length:** Average of 5.84 cm with a standard deviation of 0.83 cm, ranging from 4.3 cm to 7.9 cm.
    *   **Sepal Width:** Average of 3.06 cm with a standard deviation of 0.44 cm, ranging from 2.0 cm to 4.4 cm. This attribute shows less dispersion than sepal length.
    *   **Petal Length:** Average of 3.76 cm with the highest standard deviation (1.77 cm) and the widest range (1.0 cm to 6.9 cm), indicating significant variability.
    *   **Petal Width:** Average of 1.20 cm with a standard deviation of 0.76 cm, ranging from 0.1 cm to 2.5 cm, also showing considerable variability.
    *   Petal dimensions (length and width) generally exhibit greater variability compared to sepal dimensions, suggesting they might be more influential characteristics for distinguishing between Iris species.
*   **Categorical Feature (Species):**
    *   The dataset contains 3 unique species: 'setosa', 'versicolor', and 'virginica'.
    *   Each of the three species is equally represented with 50 instances, indicating a perfectly balanced dataset.
    *   'setosa' was identified as the most frequent species, with a frequency of 50.

### Insights or Next Steps

*   The dataset's balanced distribution of species and absence of missing values simplify the data preparation phase for machine learning tasks.
*   Given the higher variability in petal measurements, future analysis or model development should focus on these features, as they are likely to be strong predictors for classifying Iris species.
